streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -171,16 +171,17 @@ public void onCompletion(final RecordMetadata metadata,
                             if (exception instanceof ProducerFencedException) {
                                 log.warn(LOG_MESSAGE, key, value, timestamp, topic, exception.toString());
                                 sendException = new ProducerFencedException(
-                                    String.format(
-                                        EXCEPTION_MESSAGE,
-                                        logPrefix,
-                                        "producer got fenced",
-                                        key,
-                                        value,
-                                        timestamp,
-                                        topic,
-                                        exception.toString()
-                                    )
+                                        String.format(
```

Comment:
I don't thinks that this does what we want. The callback that is executed here, is not related to stores.

Note, that `sendException` is only created here, but not directly throw. It is throw in L229. Also, from my understanding, the code calling `checkForException()` might not have the knowledge about the actual Serdes. But you should double check this.

Btw: the log statement in L193 also needs a fix.